neqo-transport/src/connection.rs

```diff
@@ -470,18 +522,28 @@ impl Connection {
     }
 
     /// Get the time that we next need to be called back, relative to `now`.
-    fn next_delay(&mut self, now: Instant) -> Option<Duration> {
+    fn next_delay(&mut self, now: Instant) -> Duration {
         self.loss_recovery_state = self.loss_recovery.get_timer(&self.state);
+        let ack_time = self.acks.ack_time();
 
-        let time = match (self.loss_recovery_state.callback_time, self.acks.ack_time()) {
-            (Some(t_lr), Some(t_ack)) => Some(min(t_lr, t_ack)),
-            (Some(t), _) | (_, Some(t)) => Some(t),
-            _ => None,
+        qdebug!([self] "LR timer?: {:?} ack timer?: {:?}", self.loss_recovery_state.callback_time, ack_time);
+
+        let mut timeout = match (self.loss_recovery_state.callback_time, ack_time) {
+            (Some(t_lr), Some(t_ack)) => min(t_lr, t_ack),
+            (Some(t), _) | (_, Some(t)) => t,
+            (None, None) => self.idle_timeout.as_instant().expect(
+                "only called after something has happened that will cause \
+                 idle timeout to exist",
+            ),
         };
 
+        if let Some(idle_timeout) = self.idle_timeout.as_instant() {
+            timeout = min(timeout, idle_timeout);
+        }
```

Comment:
It seems wrong to call self.idle_timeout.as_instant() twice in the same function.

Reply:
I dislike this whole function. Once we add congestion control/pacing there will be yet another possible minimum timer duration source  (that makes...4?). I'm going to leave as-is but add a TODO to redo this whole function in the context of CC development and be more abstracted.

Reply:
nm, I just went ahead and rewrote `next_delay`, LMKWYT.